docs/source/distribution_implementation.md

```diff
@@ -0,0 +1,452 @@
+    def __call__(self, loc=0.0, scale=1.0, size=None, **kwargs) -> TensorVariable:
+        return super().__call__(loc, scale, size=size, **kwargs)
+
+    # This is the Python code that produces samples.  Its signature will always
+    # start with a NumPy `RandomState` object, then the distribution
+    # parameters, and, finally, the size.
+    #
+    # This is effectively the v4 replacement for `Distribution.random`.
+    @classmethod
+    def rng_fn(
+        cls,
+        rng: np.random.RandomState,
+        loc: np.ndarray,
+        scale: np.ndarray,
+        size: Tuple[int, ...],
+    ) -> np.ndarray:
+        return scipy.stats.blah.rvs(loc, scale, random_state=rng, size=size)
+
+# Create an instance of the `RandomVariable` `Op`...
+blah = BlahRV()
+
+```
+
+Some important things to keep in mind:
+
+1. Everything inside the `rng_fn` method is pure Python code (as are the inputs)
+   and should not make use of other `Aesara` symbolic ops. The random method
+   should make use of the `rng` which is a Numpy `Random State / Generator` ??,
+   so that samples are reproducible.
+1. The `size` argument (together with the inputs shape) is the only way for the
+   user to specify the (shape?, number of draws?) of the `RandomVariable`.
+   The `rng_fn` will have to take this into consideration for correct output.
+   `size` is the specification used by `Numpy` and `Scipy` and works like
+   PyMC3 `shape` for univariate distributions, but is different for multivariate
+   distributions. Unfortunately there is no general reference documenting how
+   `size` ought to work for multivariate distributions.
+   This [discussion](https://github.com/numpy/numpy/issues/17669) may be helpful
+   to get more context.
+1. `Aesara` tries to infer the output shape of the `RandomVariable`
+   (given a user-specified size) by introspection of the `ndim_supp` and
+   `ndim_params` attributes. However, the default method may not work for more
+   complex distributions. In that case, custom `_shape_from_params` (and less
+   probably (?) `_infer_shape`) should also be implemented in the new
+   `RandomVariable` class. One simple example is seen in the [MultinomialRV](https://github.com/aesara-devs/aesara/blob/3a3adaee3b8ceccff6d0208996f117fcd5e5eb6f/aesara/tensor/random/basic.py#L518-L521)
+   where it was necessary to specify the `rep_param_idx` so that the
+   `default_shape_from_params` helper method could do its job. In more complex
+   cases, it may not be possible to make use of the default helper,
+   but those have not been found yet!
+1. It's okay to use the `rng_fn` `classmethods` of other Aesara and PyMC3
+   `RandomVariables` inside the new `rng_fn`. For example if you are
+   implementing a negative HalfNormal `RandomVariable`, your `rng_fn` can simply
+   return `- halfnormal.rng_fn(rng, scale, size)`.
+
+
+For a quick test that your new `RandomVariable` `Op` is working, you can just
+call `eval` on a new instance:
+
+```python
```

Comment:
you have structured all the code as input and output with outputs within the python highlighted code cell, would it make sense to make the file executable? and either run it every time we build docs or save it as ipynb so output gets stored and we don't need to run every time?

Reply:
I am not sure how functional the pseudo code is :/